File: trunk/obvious/src/main/java/obvious/data/util/TableListener.java
package obvious.data.util;

import obvious.data.Table;

/**
 * Listener interface for monitoring changes to a table.
 *
 * @author <a href="http://jheer.org">jeffrey heer</a>
 */
public interface TableListener {

    int ALL_COLUMN = -1;

    int BEGIN_EDIT = -0;
    int END_EDIT = -0;

    /**
     * Notification that a table has changed.
     * @param t the table that has changed
     * @param start the starting row index of the changed table region
     * @param end the ending row index of the changed table region
     * @param col the column that has changed, or
     * {@link EventConstants#ALL_COLUMNS} if the operation affects all
     * columns
     * @param type the type of modification, one of
     * {@link EventConstants#INSERT}, {@link EventConstants#DELETE}, or
     * {@link EventConstants#UPDATE}.
     */
    void tableChanged(Table t, int start, int end, int col, int type);

} // end of interface TableListener

File: trunk/obvious-c++/src/obvious/data/TableListener.cxx


#include "obvious/data/event/TableListener.hpp"

namespace obvious {
  
  TableListener::TableListener()
  {
  }

  void TableListener::beginEdit(int)
  {
  }

  bool TableListener::endEdit(int context)
  {
	  return true;
  }

  bool TableListener::checkInvariants()
  {
	  return true;
  }

  void TableListener::tableChanged(Table *, int, int, int, int)
  {
  }

} // namespace obvious
